### PR TITLE
Add e-mail auth docs

### PR DESCRIPTION
**Pull request by infomiho**
- Added the e-mail auth guide
- Added the e-mail auth part in the features doc
- I've moved things a bit in the Authentication section of features, but didn't do any major changes.

### CODE REVIEW

File: web/docs/language/features.md [hunk 1]
```diff
@@ -954,16 +955,17 @@ app MyApp {
 `app.auth` is a dictionary with following fields:
 
 #### `userEntity: entity` (required)
-Entity which represents the user (sometimes also referred to as *Principal*).
```
Comment (infomiho) [inline web/docs/language/features.md:957]: @Martinsos agreed to remove this

File: web/docs/language/features.md [hunk 2]
```diff
@@ -1491,6 +1574,103 @@ This function should return the user fields to use when creating a new user upon
   - `generateAvailableDictionaryUsername` generates a random dictionary phrase that is not yet in the database. For example, `nice-blue-horse-27160`.
 
 
+### Validation Error Handling
```
Comment (infomiho) [inline web/docs/language/features.md:1577]: Old content

File: web/docs/language/features.md [hunk 3]
```diff
@@ -1491,6 +1574,103 @@ This function should return the user fields to use when creating a new user upon
...
+  } catch (e) {
+    if (e instanceof AuthError) {
+      throw new HttpError(422, 'Validation failed', { message: e.message })
+    } else {
+      throw e
+    }
+  }
+```
+
+## Accessing the currently logged in user
```
Comment (infomiho) [inline web/docs/language/features.md:1598]: Old content

File: web/docs/guides/email-auth.md [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+    },
+    onAuthFailedRedirectTo: "/login",
+    onAuthSucceededRedirectTo: "/profile"
+  },
+}
+```
+
+### User entity
+
+Then we'll defined the `User` entity in our `main.wasp` file:
```
Comment (shayneczyzewski) [inline web/docs/guides/email-auth.md:80]:
```suggestion
Then we'll define the `User` entity in our `main.wasp` file:
```

File: web/docs/guides/email-auth.md [hunk 5]
```diff
@@ -0,0 +1,426 @@
...
+    <div>
+      Hello, {user.email}!
+    </div>
+  )
+}
+```
+
+## Conclusion
+
+And that's it! We now have a full authentication system in our app. We can register new users, login, log-out, verify their e-mail, and reset their password.
```
Comment (shayneczyzewski) [inline web/docs/guides/email-auth.md:424]: I'd unify login and logout somehow, either no dashes for both, etc.

**Review by shayneczyzewski — APPROVED** [review]
Looks good to me, nice job @infomiho! I had two thoughts you can use or not:
1) Should we also mention any caveats with using email with social providers (like if they try to sign up but they already signed in with a Google email)? On the fence here, probably lean no but wanted to raise.
2) Should we update the ChangeLog with a breaking change around the form inputs going from default to named: `import { SignupForm } from '@wasp/auth/forms/Signup'`? I would recommend we do this one.

File: web/docs/guides/email-auth.md [hunk 6]
```diff
@@ -0,0 +1,426 @@
+---
+title: Email Authentication
+---
+
+# Email Authentication
+
+## Overview
+
+Wasp supports e-mail authentication out of the box. It provides a set of routes and email templates that you can use to implement email authentication in your app.
```
Comment (matijaSos) [inline web/docs/guides/email-auth.md:9]:
```suggestion
Wasp supports e-mail authentication out of the box, along with email verification and "forgot your password?" flows. It provides a set of routes and email templates that you can use to implement it in your app.
```

File: web/docs/guides/email-auth.md [hunk 7]
```diff
@@ -0,0 +1,426 @@
...
+
+![Auth UI](/img/authui/all_screens.gif)
+
+In this guide, we'll go through the easiest way to set up email authentication: using Wasp's Auth UI components.
+
+## Outline of the guide
+
+We'll need to do the following steps to set up email authentication:
```
Comment (matijaSos) [inline web/docs/guides/email-auth.md:17]:
```suggestion
We'll need to take the following steps to set up email authentication:
```

File: web/docs/guides/email-auth.md [hunk 8]
```diff
@@ -0,0 +1,426 @@
...
+- [ ] Set up email authentication in `main.wasp`
+- [ ] Define the user entity
```
Comment (matijaSos) [inline web/docs/guides/email-auth.md:19]:
```suggestion
- [ ] Add the user entity
```

File: web/docs/guides/email-auth.md [hunk 9]
```diff
@@ -0,0 +1,426 @@
...
+- [ ] Define the routes and pages
```
Comment (matijaSos) [inline web/docs/guides/email-auth.md:20]:
```suggestion
- [ ] Add the routes and pages
```

File: web/docs/guides/email-auth.md [hunk 10]
```diff
@@ -0,0 +1,426 @@
...
+- [ ] Set up the email sender in `main.wasp` and `.env.server`
+- [ ] Use Auth UI in our pages
```
Comment (matijaSos) [inline web/docs/guides/email-auth.md:22]:
```suggestion
- [ ] Use Auth UI components in our pages
```

**Review by matijaSos — APPROVED** [review]
lgtm! a couple of suggestions. we can probably simplify this even more in the future but this is a great start. Plus the gifs rock :D

File: web/docs/guides/email-auth.md [hunk 11]
```diff
@@ -0,0 +1,426 @@
...
+
+We are using the `SignupForm` component from `@wasp/auth/forms/Signup` to render the signup form.
+
+```tsx title="client/pages/auth/Signup.tsx"
+import { Link } from 'react-router-dom'
+import { SignupForm } from '@wasp/auth/forms/Signup'
+
+export function Signup () {
+  return (
+    <div className="w-full h-full bg-white">
```
[the remark this answers is not in the document]
Comment (infomiho) [reply web/docs/guides/email-auth.md:165]: I did it somewhere :)

[the remark this answers is not in the document]
Comment (infomiho) [reply web/docs/guides/email-auth.md:165]: https://github.com/wasp-lang/wasp/pull/1113/files#diff-597d2bdfcf259abbdcd280fbb7f2140ca7a0fd76157ffa630868082ee3bcb7ffR269

[the remark this answers is not in the document]
Comment (infomiho) [reply web/docs/guides/email-auth.md:165]: Screenshot 2023-04-07 at 16 19 13

Added an info box

**Comment by infomiho** [comment]
@shayneczyzewski I've updated the docs about social + email clashes and Changelog with the breaking changes

@matijaSos I've updated the texts

**Pull request merged**